lib/ruby_smb/smb1/packet/nt_create_andx_request.rb

```diff
@@ -0,0 +1,58 @@
+module RubySMB
+  module SMB1
+    module Packet
+
+      # A SMB1 SMB_COM_NT_CREATE_ANDX Request Packet as defined in
+      # [2.2.4.64.1 Request](https://msdn.microsoft.com/en-us/library/ee442175.aspx) and
+      # [2.2.4.9.1 Client Request Extensions](https://msdn.microsoft.com/en-us/library/cc246332.aspx)
+      class NtCreateAndxRequest < RubySMB::GenericPacket
+
+        # A SMB1 Parameter Block as defined by the {NtCreateAndxRequest}
+        class ParameterBlock < RubySMB::SMB1::ParameterBlock
+          endian :little
+          struct :words, onlyif: -> { word_count.nonzero? } do
+            uint8                              :andx_command,        label: 'AndX Command',       initial_value: 0xFF
+            uint8                              :andx_reserved,       label: 'AndX Reserved',      initial_value: 0x00
+            uint16                             :andx_offset,         label: 'AndX Offset',        initial_value: 0x0000
```

Comment:
According to the documentation:

> This field MUST be set to the offset in bytes from the start of the SMB Header (section 2.2.3.1) to the start of the WordCount field in the next SMB command in this packet. This field is valid only if the AndXCommand field is not set to 0xFF. If AndXCommand is 0xFF, this field MUST be ignored by the server.

This logic has not been implemented yet. Do you think it would be better to add this here (in a "value:" attribute for example), or in the client when SMB commands are chained (Andx Messages)?

Reply:
There is already an andx_block field defined at https://github.com/rapid7/ruby_smb/blob/master/lib/ruby_smb/smb1/andx_block.rb that handles everything for the AndX

Reply:
Thank you for the heads-up, I really missed this one. I will fix this.

Reply:
Addressed in 07d5066e9de72b9dce9101cf62967e1d3fde9833.